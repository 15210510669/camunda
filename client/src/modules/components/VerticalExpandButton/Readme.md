VerticalExpandButton:

```js
<div style={{height: '500px', width: '100px'}}>
  <VerticalExpandButton label="Selections">
    <button>test</button>
  </VerticalExpandButton>
</div>
```
