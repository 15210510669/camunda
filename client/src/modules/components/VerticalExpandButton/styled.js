import styled from 'styled-components';
import {themed, Colors, themeStyle} from 'modules/theme';

export const Button = themed(styled.button`
  height: 100%;
  width: 56px;
  padding: 11px;

  background: ${themeStyle({
    dark: Colors.uiDark03,
    light: Colors.uiLight02
  })};
  color: ${themeStyle({
    dark: '#fff',
    light: Colors.uiLight06
  })};
  border: none;

  opacity: 0.9;
  font-size: 15px;
  font-weight: bold;

  position: relative;
`);

export const Vertical = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  transform: rotate(-90deg) translateX(-100%) translateY(100%);
  transform-origin: 0 0;
  display: flex;
  align-items: center;
  margin-top: 11px;
`;
