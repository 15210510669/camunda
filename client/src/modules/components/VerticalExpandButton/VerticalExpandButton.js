import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled';

export default function VerticalExpandButton({label, children, ...otherProps}) {
  return (
    <Styled.Button title={`Expand ${label}`} {...otherProps}>
      <Styled.Vertical>
        <Styled.Label>{label}</Styled.Label>
        {children}
      </Styled.Vertical>
    </Styled.Button>
  );
}

VerticalExpandButton.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node
};
