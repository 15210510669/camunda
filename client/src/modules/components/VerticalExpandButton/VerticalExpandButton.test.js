import React from 'react';
import {shallow} from 'enzyme';

import VerticalExpandButton from './VerticalExpandButton';

describe('VerticalExpandButton', () => {
  it('should render a button', () => {
    const label = 'Random label';
    const children = <button>Child content</button>;
    const node = shallow(
      <VerticalExpandButton label={label}>{children}</VerticalExpandButton>
    );

    expect(node.props().title).toBe(`Expand ${label}`);

    expect(node).toMatchSnapshot();
  });
});
