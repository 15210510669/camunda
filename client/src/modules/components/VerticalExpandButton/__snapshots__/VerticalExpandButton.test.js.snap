// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalExpandButton should render a button 1`] = `
<Themed(styled.button)
  title="Expand Random label"
>
  <styled.span>
    <styled.span>
      Random label
    </styled.span>
    <button>
      Child content
    </button>
  </styled.span>
</Themed(styled.button)>
`;
